### PR TITLE
SED-3564 improve-time-selector

### PR DESCRIPTION
**Pull request by neogucky**
Solution for requesting specific time range from aggregated tree endpoint

### CODE REVIEW

File: step-plans/step-plans-core/src/main/java/step/core/artefacts/reports/aggregated/AggregatedReportViewBuilder.java
Comment (cl-exense) [inline step-plans/step-plans-core/src/main/java/step/core/artefacts/reports/aggregated/AggregatedReportViewBuilder.java:38]: Looks correct. Here's an equivalent solution that avoids explicit branches. I'll leave it up to you to decide which one you prefer and find more readable, I'd be fine with both.
```
        return Optional.ofNullable(execution.getResolvedPlanRootNodeId())
                .map(resolvedPlanNodeAccessor::get)
                .map(node -> recursivelyBuildAggregatedReportTree(node, request))
                .orElse(null);
```

**Review by cl-exense — APPROVED** [review]
Approving with optional alternative suggestion

File: step-plans/step-plans-core/src/main/java/step/core/artefacts/reports/aggregated/AggregatedReportViewBuilder.java
Comment (david-stephan) [reply step-plans/step-plans-core/src/main/java/step/core/artefacts/reports/aggregated/AggregatedReportViewBuilder.java:39]: feedback implemented

**Pull request merged**
